compiler/src/dotty/tools/backend/jvm/BCodeBodyBuilder.scala

```diff
@@ -826,61 +827,170 @@ trait BCodeBodyBuilder extends BCodeSkelBuilder {
       generatedType
     }
 
-    /*
-     * A Match node contains one or more case clauses,
-     * each case clause lists one or more Int values to use as keys, and a code block.
-     * Except the "default" case clause which (if it exists) doesn't list any Int key.
-     *
-     * On a first pass over the case clauses, we flatten the keys and their targets (the latter represented with asm.Labels).
-     * That representation allows JCodeMethodV to emit a lookupswitch or a tableswitch.
-     *
-     * On a second pass, we emit the switch blocks, one for each different target.
+    /* A Match node contains one or more case clauses, each case clause lists one or more
+     * Int/String values to use as keys, and a code block. The exception is the "default" case
+     * clause which doesn't list any key (there is exactly one of these per match).
      */
     private def genMatch(tree: Match): BType = tree match {
```

Comment:
Note that the bytecode this generates ends up having a number of bogus `goto,nop,..,nop,athrow` sequences in it. Scala 2 doesn't have these because [it does some extra dead-code elimination](https://github.com/scala/scala/pull/8936#issuecomment-809211979) to clean up junk like this (the extra instructions are produced by `ClassWriter`). I hope to get around to that at some point, but likely not in this PR.